### PR TITLE
Return all releases for a release group

### PR DESCRIPTION
**Pull request by ta264**
Metadata service update to go with my DB schema rework

### CODE REVIEW

File: lidarrmetadata/api.py [hunk 1]
```diff
@@ -82,7 +82,21 @@ def default_route():
...
-def get_artist_info(mbid):
+def get_artist_info_route(mbid):
+
+    output = get_artist_info(mbid,
+                             True,
+                             request.args.get('primTypes', None),
+                             request.args.get('secTypes', None),
+                             request.args.get('releaseStatuses', None))
+
+    if (type(output) is dict):
```
Comment (danielunderwood) [inline lidarrmetadata/api.py]: Probably more typical to use `isinstance(output, dict)` here.

File: lidarrmetadata/api.py [hunk 2]
```diff
@@ -120,9 +127,16 @@ def get_artist_info(mbid):
...
             artist['Links'])
+        wikidata_links = filter(
+            lambda link: 'wikidata' in link.get('target', ''),
+            artist['Links'])
+
         if wikipedia_links:
             artist['Overview'] = overview_providers[0].get_artist_overview(
                 wikipedia_links[0]['target'])
+        elif wikidata_links:
+            artist['Overview'] = overview_providers[0].get_artist_overview(
```
Comment (danielunderwood) [inline lidarrmetadata/api.py:148]: This may handle #64.

File: lidarrmetadata/provider.py [hunk 3]
```diff
@@ -759,31 +564,37 @@ def get_album_media(self, album_id):
...
                  'Type': result['primary_type'],
                  'SecondaryTypes': result['secondary_types'],
                  'ReleaseStatuses': result['release_statuses'],
                  'ReleaseDate': datetime.datetime(result['year'] or 1,
                                                   result['month'] or 1,
-                                                  result['day'] or 1)}
+                                                  result['day'] or 1),
+                 'Rating': {
+                     'Count': result['rating_count'],
+                     'Value': (result['rating'] or 0) / 10 or None
```
Comment (danielunderwood) [inline lidarrmetadata/provider.py]: I was thinking that there was some reason that I wrote it like this for the artist rating, but `0 / 10 or None` will return `None` in the case of the rating being `0`. Maybe something like `result['rating']  / 10 if result['rating'] is not None else None`?

File: lidarrmetadata/provider.py [hunk 4]
```diff
@@ -370,203 +391,16 @@ def search_artist(self, name):
                 for result in results]
 
 
-class MusicbrainzApiProvider(Provider,
```
Comment (danielunderwood) [inline lidarrmetadata/provider.py:387]: Good catch. I had been meaning to get rid of this code.

**Review by danielunderwood — COMMENTED** [review]
Looks pretty good. Let me see if I can get this deploying to the development endpoint of our server.

**Comment by danielunderwood** [comment]
Also note that you'll want to bump the version like in 9836c25. The branch that commit is in it up at https://api.lidarr.audio/api/testing/, though I'll need to pull any additional changes you make for it to build.

File: lidarrmetadata/provider.py [hunk 5]
```diff
@@ -936,17 +748,26 @@ def __init__(self):
         self._client = mediawikiapi.MediaWikiAPI()
 
     def get_artist_overview(self, url):
```
Comment (danielunderwood) [inline lidarrmetadata/provider.py:799]: Looks like there's a `NameError: global name 'url' is not defined` here when requesting https://api.lidarr.audio/api/testing/artist/93bd1037-d736-42ca-93c3-c397f2cb1165, though it's fine on other artists. I don't see anything obvious wrong here, but the Sentry stack trace only shows the Flask calls. May be worth checking out that artist with your dev instance.

Comment (danielunderwood) [reply lidarrmetadata/provider.py:799]: Sorry for all the comments, but looks like this is also at least part of the test failures. See https://travis-ci.org/lidarr/LidarrAPI.Metadata/jobs/448969259.

**Comment by ta264** [comment]
I've made the changes you suggested and fixed the tests.  It was the logger, which didn't like unicode in the wikipedia title name.

File: lidarrmetadata/api.py [hunk 6]
```diff
@@ -90,7 +90,7 @@ def get_artist_info_route(mbid):
                              request.args.get('secTypes', None),
                              request.args.get('releaseStatuses', None))
 
-    if (type(output) is dict):
+    if (isinstance(output, dict)):
```
Comment (danielunderwood) [inline lidarrmetadata/api.py]: Can also pull out the extra parentheses here.

File: lidarrmetadata/provider.py [hunk 7]
```diff
@@ -936,27 +748,39 @@ def __init__(self):
...
         :param url: URL of wikipedia page
         :return: Summary String
         """
         try:
-            title = self.title_from_url(url)
             return self._client.summary(title, auto_suggest=False)
         # FIXME Both of these may be recoverable
         except mediawikiapi.PageError as error:
-            logger.error('Wikipedia PageError from {url}: {e}'.format(e=error.message, url=url))
+            logger.error(u'Wikipedia PageError from {title}: {e}' % {'e':error, 'title':title})
```
Comment (ta264) [inline lidarrmetadata/provider.py:826]: I could only get this to work correctly if I used the old-fashioned `%` instead of `.format`.  I have no idea why.

Comment (danielunderwood) [reply lidarrmetadata/provider.py:826]: It'll probably be fine for now. I'm planning to update everything for py3 when I get a chance and will likely change these to f-strings.

File: lidarrmetadata/api.py (hunk 2 above)
Comment (Qstick) [reply lidarrmetadata/api.py:148]: Should we flip the order?

**Pull request merged**